crates/matrix-sdk/src/authentication/oauth/account_management_url.rs

```diff
@@ -73,119 +73,174 @@ pub enum AccountManagementActionFull {
     CrossSigningReset,
 }
 
-/// Build the URL for accessing the account management capabilities, as defined
-/// in [MSC].
+/// Builder for the URL for accessing the account management capabilities, as
+/// defined in [MSC4191].
 ///
-/// # Arguments
+/// This type can be instantiated with [`OAuth::account_management_url()`] and
+/// [`OAuth::fetch_account_management_url()`].
 ///
-/// * `account_management_uri` - The URL to access the issuer's account
-///   management capabilities.
-///
-/// * `action` - The action that the user wishes to take.
-///
-/// # Returns
-///
-/// A URL to be opened in a web browser where the end-user will be able to
-/// access the account management capabilities of the issuer.
-///
-/// # Errors
-///
-/// Returns an error if serializing the action fails.
+/// [`AccountManagementUrlBuilder::build()`] and
+/// [`AccountManagementUrlBuilder::build_or_ignore_action()`] return a URL to be
+/// opened in a web browser where the end-user will be able to access the
+/// account management capabilities of the issuer.
```

Comment:
Can we get an example here or on the method that returns this struct? There's now enough indirection for this to warrant some examples.

Reply:
I added an example.